lib/viewmodels/map/map_view_model.dart

```diff
@@ -0,0 +1,36 @@
+import "dart:async";
+
+import "package:google_maps_flutter/google_maps_flutter.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/services/geolocation_service.dart";
+import "package:proxima/viewmodels/map/map_state.dart";
+
+final mapNotifierProvider = StateNotifierProvider<MapController, MapState>(
+  (ref) => MapController(ref),
+);
+
+class MapController extends StateNotifier<MapState> {
```

Comment:
`StateNotifier` is deprecated. We should only use `Notifier` or `AsyncNotifier`.
Also is this PR about the view or the viewmodel because I feel like we are mixing two things here.

Reply:
@yoannLafore made a similar comment

Reply:
The viewmodel of the map now uses an AsyncNotifierProvider.